### ISSUE TITLE
New user: ArgumentError: 'nothing' should not be printed

### ISSUE THREAD

**Issue by BioTurboNick** [JuliaData/CSV.jl#352]
I'm just getting started with Julia and trying to use CSV.jl to read a tab-delimited file.

I'm trying to iterate over the rows, but anything I do with them results in "ArgumentError: `nothing` should not be printed; use `show`, `repr`, or custom output instead."

I have no idea where "nothing" would be entering into the data, as all rows and columns are populated in the source file.  And unless I misunderstand the interface, there's no other way to view the data in the structure to inspect what is happening? Any way to pick out a row or column or entry?

**Comment by quinnj** [comment]
Welcome! Could you share your code leading to the problem? That should help figuring out what is going wrong where.

**Comment by BioTurboNick** [comment]
```
file = CSV.File(path)
for row ∈ file
    println(row.X)
end
```
There's a column with a name of X, which appears in the Symbol() listing for Tables.Schema. I'm just assuming that the access name is also "X", though not sure how it handles column names with spaces like "Channel Name"...

Oh, here's the stack trace:

```
 [1] print(::Base.GenericIOBuffer{Array{UInt8,1}}, ::Nothing) at .\show.jl:566
 [2] print_to_string(::String, ::Vararg{Any,N} where N) at .\strings\io.jl:122
 [3] string at .\strings\io.jl:155 [inlined]
 [4] to_index(::Nothing) at .\indices.jl:265
 [5] to_index(::Array{Type,1}, ::Nothing) at .\indices.jl:247
 [6] to_indices at .\indices.jl:293 [inlined]
 [7] to_indices at .\indices.jl:290 [inlined]
 [8] getindex at .\abstractarray.jl:905 [inlined]
 [9] getproperty(::CSV.UntypedRow{CSV.File{false,true,Base.GenericIOBuffer{Array{UInt8,1}},Parsers.Delimited{false,Parsers.Quoted{Parsers.Strip{Parsers.Sentinel{typeof(Parsers.defaultparser),Parsers.Trie{0x00,false,missing,2,Tuple{}}}}},Parsers.Trie{0x00,false,missing,8,Tuple{Parsers.Trie{0x2c,true,missing,8,Tuple{}},Parsers.Trie{0x0a,true,missing,8,Tuple{}},Parsers.Trie{0x0d,true,missing,8,Tuple{Parsers.Trie{0x0a,true,missing,8,Tuple{}}}}}}},NamedTuple{(),Tuple{}}}}, ::Symbol) at C:\Users\nicho\AppData\Local\JuliaPro-1.0.1.1\pkgs-1.0.1.1\packages\CSV\uLyo0\src\tables.jl:148
 [10] top-level scope at .\none:3
```

**Comment by BioTurboNick** [comment]
Ohhhhh. Duh.

I did not realize that the function required the delimiter to be specified. Maybe this issue should be turned into a suggestion to make it easier to understand when input is not read correctly. Or documentation to clarify what's required a bit more? Idk.

I saw
```
Tables.Schema:
 Symbol("Channel Name\tX\tY\tXc\tYc\tHeight\tArea\tWidth\tPhi\tAx\tBG\tI\tFrame\tLength\tLink\tValid\tZ\tZc\tPhotons\tLateral Localization Accuracy\tXw\tYw\tXwc\tYwc\tZw\tZwc")  …  Union{Missing, String}
```

And just assumed that was normal. Julia can be hard to follow in its output sometimes. Thanks, I think it works now haha.

**Comment by BioTurboNick** [comment]
Although, even if it was reading it all as one column, I suppose it shouldn't have produced the error it did?

**Comment by BioTurboNick** [comment]
I guess the real error is that I was trying to access a property that didn't exist in the schema, which I suppose is an issue with Tables.jl. I'll post an issue there then.

**Comment by quinnj** [comment]
This is actually a CSV.jl issue; more helpful error message incoming.
